src/main.c

```diff
@@ -142,10 +148,117 @@ static bool is_valid_stream(char *s)
 	return true;
 }
 
+char* get_saved_file_path(void)
+{
+	char* home_path = getenv("XDG_STATE_HOME");
+
+	if (home_path == NULL)
+	{
+		// fallback to $HOME/.local/state
+		home_path = getenv("HOME");
+
+		const char* local_path = "/.local/state";
+		size_t len_home_path = strlen(home_path);
+		size_t len_local_path = strlen(local_path);
+
+		char* fallback_path = malloc(len_home_path + len_local_path + 1);
```

Comment:
You are allocating memory here but never freeing it.
Since you are using home_path one time for getenv and another time with malloc you'll have to find a way to free it one time and another time not. Or you strdup() the getenv result.